amethyst_renderer_new/src/graph/pass.rs

```diff
@@ -0,0 +1,188 @@
+
+//! Pass tells renderer how to convert inputs to image.
+//!
+//! # Definition
+//! Pass - is a box which get some:
+//!
+//! * Inputs
+//!   * Attachments
+//!   * Meshes combined with:
+//!     * Sampled images
+//!     * Uniform buffers
+//!     * Push constants
+//! * Outputs
+//!  * Attachments
+//!  * Results of queries (Let's forget about it for now)
+//!
+//! But it gets executed on the GPU.
+//! So instead of writing it as a function and calling with arguments needed
+//! we have to record commands into buffers and send them to the GPU.
+//!
+//! We want a way to define a box which will record all necessarry commands in declarative fasion.
+//! In order to feed this box with data we also need define `World -> [Input]` conversion
+//! (in declarative fasion where possible).
+//!
+//!
+
+use std::fmt::{self, Debug};
+use std::marker::PhantomData;
+
+use core::Transform;
+use gfx_hal::Backend;
+use gfx_hal::format::Format;
+use gfx_hal::memory::cast_slice;
+use gfx_hal::pso::{DescriptorSetLayoutBinding, GraphicsShaderSet, PipelineStage};
+use gfx_hal::queue::capability::{Supports, Transfer};
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use rayon_core::current_thread_index;
+use specs::{SystemData, World};
+
+use vertex::VertexFormat;
+
+error_chain!{}
+
+pub trait Data<'a, B>
+where
+    B: Backend,
```

Comment:
It is necessary so that pass can implement it differently for different backend. Look at `Flat`. Another way is to have backend type as type parameter for pass. Not sure which is better

Reply:
That still doesn't make it necessary for the trait to declare that bound.